cclib/method/ddec.py

```diff
@@ -212,32 +273,33 @@ def calculate_refcharges(self):
             )
             closest_r_index = numpy.abs(self.radial_grid_r[atomi] - dist_r).argmin()
 
+            # Equation 54 in doi: 10.1039/c6ra04656h
             stockholder_w[atomi][xindex][yindex][zindex] = self.proatom_density[atomi][
                 closest_r_index
             ]
 
+        # Equation 55 in doi: 10.1039/c6ra04656h
         localized_w = numpy.power(stockholder_w, 4)
 
+        # Equation 53 in doi: 10.1039/c6ra04656h
         stockholder_bigW = numpy.sum(stockholder_w, axis=0)
         localized_bigW = numpy.sum(localized_w, axis=0)
 
-        self.logger.info("Creating first reference charges.")
-        self.refcharges = numpy.zeros((atoms))
-        self._localizedcharges = numpy.zeros((atoms))
-        self._stockholdercharges = numpy.zeros((atoms))
+        refcharges = numpy.zeros((atoms))
+        localizedcharges = numpy.zeros((atoms))
+        stockholdercharges = numpy.zeros((atoms))
 
         for atomi in range(atoms):
-            chargedensity = copy.deepcopy(self.chgdensity)
-            localized_weight = localized_w[atomi] / localized_bigW
-            chargedensity.data *= localized_weight
-            self._localizedcharges[atomi] = self.data.atomnos[atomi] - chargedensity.integrate()
-
-            chargedensity = copy.deepcopy(self.chgdensity)
-            stockholder_weight = stockholder_w[atomi] / stockholder_bigW
-            chargedensity.data *= stockholder_weight
-            self._stockholdercharges[atomi] = self.data.atomnos[atomi] - chargedensity.integrate()
+            # Equation 52 and 51 in doi: 10.1039/c6ra04656h
```

Comment:
It took me a second to realize that their `Stock/Loc` superscript means Stockholder or localized (`{Stock,Loc}`), not one divided by the other...

Reply:
Yes -- the notations in the paper puzzled me for a while too!